packages/devtools_server/lib/src/memory_profile.dart

```diff
@@ -0,0 +1,311 @@
+        print('Error [MEMORY_PROTOCOL]: $e');
+        return null;
+      }
+    }));
+
+    // Polls for current Android meminfo using:
+    //    > adb shell dumpsys meminfo -d <package_name>
+    final isolate = isolates[0];
+    _selectedIsolate =
+        IsolateRef(id: isolate.id, name: isolate.name, number: isolate.number);
+
+    if (hasConnection &&
+        vm.operatingSystem == 'android' &&
+        _selectedIsolate != null) {
+      // Poll ADB meminfo
+      adbMemoryInfo = await _fetchAdbInfo();
+    } else {
+      // TODO(terry): TBD alternative for iOS memory info - all values zero.
+      adbMemoryInfo = AdbMemoryInfo.empty();
+    }
+
+    // Polls for current RSS size.
+    _update(vm, isolates);
+
+    _pollingTimer = Timer(kUpdateDelay, _pollMemory);
+  }
+
+  Future<AdbMemoryInfo> _fetchAdbInfo() async =>
+      AdbMemoryInfo.fromJson((await getAdbMemoryInfo()).json);
+
+  void _update(VM vm, List<Isolate> isolates) {
+    processRss = vm.json['_currentRSS'];
+
+    isolateHeaps.clear();
+
+    for (Isolate isolate in isolates) {
+      if (isolate != null) {
+        final List<HeapSpace> heaps = getHeaps(isolate).toList();
+        isolateHeaps[isolate.id] = heaps;
+      }
+    }
+
+    _recalculate();
+  }
+
+  void _updateGCEvent(String id, List<HeapSpace> heaps) {
+    isolateHeaps[id] = heaps;
+    _recalculate(true);
+  }
+
+  void _recalculate([bool fromGC = false]) {
+    int total = 0;
+
+    int used = 0;
+    int capacity = 0;
+    int external = 0;
+    for (List<HeapSpace> heaps in isolateHeaps.values) {
+      used += heaps.fold<int>(0, (int i, HeapSpace heap) => i + heap.used);
+      capacity +=
+          heaps.fold<int>(0, (int i, HeapSpace heap) => i + heap.capacity);
+      external +=
+          heaps.fold<int>(0, (int i, HeapSpace heap) => i + heap.external);
+
+      capacity += external;
+
+      total += heaps.fold<int>(
+          0, (int i, HeapSpace heap) => i + heap.capacity + heap.external);
+    }
+
+    heapMax = total;
+
+    final time = DateTime.now().millisecondsSinceEpoch;
+    final sample = HeapSample(
+      time,
+      processRss,
+      capacity,
+      used,
+      external,
+      fromGC,
+      adbMemoryInfo,
+    );
+
+    if (_verboseMode) {
+      final intl.DateFormat mFormat = intl.DateFormat('hh:mm:ss.mmm');
+      final timeCollected =
+          mFormat.format(DateTime.fromMillisecondsSinceEpoch(time));
+
+      print(' Collected Sample: [$timeCollected] capacity=$capacity, '
+          'ADB MemoryInfo total=${adbMemoryInfo.total}${fromGC ? ' [GC]' : ''}');
+    }
+
+    _jsonFile.writeSample(sample);
+  }
+
+  static Iterable<HeapSpace> getHeaps(Isolate isolate) {
+    if (isolate != null) {
+      final Map<String, dynamic> heaps = isolate.json['_heaps'];
+      return heaps.values.map((dynamic json) => HeapSpace.parse(json));
+    }
+    return const Iterable.empty();
+  }
+}
+
+class JsonFile {
```

Comment:
nit: JsonFile --> MemorySnapshotFile or JsonMemorySnapshotFile or similar.

Reply:
Will use MemoryJsonFile because Snapshot means something different for memory.